OpenRA.Mods.Common/Orders/GuardOrderGenerator.cs

```diff
@@ -28,29 +28,30 @@ protected override IEnumerable<Order> OrderInner(World world, CPos xy, MouseInpu
 				yield break;
 
 			var target = FriendlyGuardableUnits(world, mi).FirstOrDefault();
-			if (target == null || Subjects.All(s => s.IsDead))
+			if (target == null)
 				yield break;
 
 			world.CancelInputMode();
 
 			var queued = mi.Modifiers.HasModifier(Modifiers.Shift);
-			yield return new Order(OrderName, null, Target.FromActor(target), queued, null, Subjects.Where(s => s != target).ToArray());
+			yield return new Order(OrderName, null, Target.FromActor(target), queued, null, subjects.Where(s => s != target).ToArray());
 		}
 
-		public override void Tick(World world)
+		public override void SelectionChanged(World world, IEnumerable<Actor> selected)
 		{
-			if (Subjects.All(s => s.IsDead || !s.Info.HasTraitInfo<GuardInfo>()))
+			subjects = selected.Where(s => s.Info.HasTraitInfo<GuardInfo>());
+			if (!subjects.Any(s => s.Info.HasTraitInfo<AutoTargetInfo>()))
```

Comment:
Assigning subjects based on `GuardInfo` and then checking `AutoTargetInfo` here looks like a typo. If this is intentional, can we have a comment explaining that?

Reply:
`Guard` uses `AttackMove` which doesn't work without `AutoTarget`. It's written like this because of
>
[12:01:43] | (abcdefg30)  we have Guard and AttackMove on a bunch of actors that can't activate  those because the command bar requires them to have AutoTarget as well
[12:02:01] | (abcdefg30) but you can activate them when you select them together with another actor that has the trait + AutoTarget
[12:02:16] | (pchote) iirc the idea was so that they would follow along if in a group with others

For which I later filed #18004.

Tl;dr: This is to keep current behaviour.